frontend/discourse/app/lib/animation-utils.js

```diff
@@ -0,0 +1,40 @@
+import { waitForPromise } from "@ember/test-waiters";
+import { isTesting } from "discourse/lib/environment";
+import { prefersReducedMotion } from "discourse/lib/utilities";
+
+export async function waitForAnimationEnd(element) {
+  return new Promise((resolve) => {
+    const style = window.getComputedStyle(element);
+    const duration = parseFloat(style.animationDuration) * 1000 || 0;
+    const delay = parseFloat(style.animationDelay) * 1000 || 0;
+    const totalTime = duration + delay;
+
+    const handleAnimationEnd = () => {
+      clearTimeout(timeoutId);
+      element.removeEventListener("animationend", handleAnimationEnd);
+      resolve();
+    };
+
+    const timeoutId = setTimeout(
+      () => {
+        element.removeEventListener("animationend", handleAnimationEnd);
+        resolve();
+      },
+      // The timeout acts as a fallback in case the "animationend" event does not fire (e.g., when no animation is present).
+      Math.max(totalTime + 50, 50)
+    );
+
+    element.addEventListener("animationend", handleAnimationEnd);
+  });
+}
```

Comment:
This is essentially the same as:

https://github.com/discourse/discourse/blob/7dfeb504990ad6677608f7b25e4f3977ffe8a7cf/frontend/discourse/app/components/d-modal.gjs#L263-L287

Can we safely replace the one in d-modal with your new utility method?